src/components/whole-app/sidebar/nav-page-links/nav-page-links.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { NavLink } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExchangeAlt, faFolder, faHome } from '@fortawesome/free-solid-svg-icons';
+import { Nav } from 'react-bootstrap';
+import styles from './nav-page-links.module.css';
+import { useEffect } from 'react';
+
+const NavPageLinks: React.FC = () => {
+  useEffect(() => {
+    const rows = document.getElementsByClassName(styles.row);
+    Array.from(rows).forEach((row) => {
+      const current = row as HTMLElement;
+      if (
+        Array.from(row.children).includes(
+          Array.from(document.getElementsByClassName(styles.active))[0]
+        ) ||
+        current.matches(':hover')
+      ) {
+        current.style.backgroundColor = 'white';
+      } else {
+        current.style.backgroundColor = 'transparent';
+      }
+    });
+  });
```

Comment:
this doesn't look right to me. generally speaking, there's always a way to avoid ever having to directly interact with the document api when you're using a modern JS framework like React. I'm not exactly sure what you're achieving here, but here's some suggestions based on my best guess. React has a whole slew of different event triggers that you should examine taking advantage of. Things like `onMouseEnter` and `onMouseLeave` might be most relevant. If you really need to, you can access the page that the app is current at using the `react-dom` library's `useHistory()` hook. I would also conjecture that you could store style options in a local `useState()` that would be a better way of have an information source because it will integrate better with React's smart re-rendering based on state changes.

Reply:
It might even be possible with just css if it's just changing style based on hover with the `:hover` selector.

Reply:
The css hover stopped working when I added the useEffect, so I was messing around and trying something else.

Reply:
There is no parent selector in css, so I tried using the Dom methods to color the background of the currently-active link button.